howso/client/base.py

```diff
@@ -201,6 +213,33 @@ def get_trainee_session_indices(self, trainee_id, session) -> Union[Index, List[
     def get_trainee_session_training_indices(self, trainee_id, session) -> Union[Index, List[int]]:
         """Get list of all session training indices for a specified session."""
 
+    @abstractmethod
+    def get_hierarchy(self, trainee_id) -> Dict:
+        """Output the hierarchy for a trainee."""
+
+    @abstractmethod
+    def rename(
+            self,
+            trainee_id, *,
+            new_name=None,
+            child_name_path=None,
+            child_id=None
+    ) -> None:
+        """Renames a contained child trainee in the hierarchy."""
+
+    @abstractmethod
+    def execute(
+            self,
+            trainee_id,
+            method, *,
```

Comment:
Put the * on its own line so its more clear where the positional end and kwarg-only start